test/test_ops.py

```diff
@@ -782,6 +811,12 @@ def test_autocast(self, iou, dtype):
         with torch.cuda.amp.autocast():
             self.test_nms_gpu(iou=iou, dtype=dtype, device="cuda")
 
+    @pytest.mark.parametrize("iou", (0.2, 0.5, 0.8))
+    @pytest.mark.parametrize("dtype", (torch.float, torch.bfloat16))
+    def test_autocast_cpu(self, iou, dtype):
+        with torch.cpu.amp.autocast():
+            self.test_nms_cpu(iou=iou, dtype=dtype)        
```

Comment:
Would it be enough to just call `test_nms_ref()` here (perhaps with slight modifications) ?

If we really need a specific test as in `test_nms_cpu` then maybe we can just inline it below instead of having `test_nms_cpu` as a standalone.

Reply:
I modified `test_nms_ref` to check the case where the `dtype` is `torch.bfloat16` on cpu. Could you please check if it looks good to you?